File: Arrays/q2.py
import sys

def isPermutation(s, t):
	if not len(s) == len(t):
		return False
	d = [0]*128
	for c in s:
		d[ord(c)] += 1
	for c in t:
		d[ord(c)] -= 1
		if d[ord(c)] < 0:
			return False
	return True

def main():
	s = raw_input().strip()
	t = raw_input().strip()
	if isPermutation(s,t):
		print "YES"
	else:
		print "NO"

main()

File: Arrays/q2.java
import java.io.*;
import java.util.*;

public class q2{
	private static boolean isPermutation(String s, String t){
		if(s.length() != t.length()){
			return false;
		}
		int[] lcount = new int[128];
		for(int i = 0;i < s.length();i++){
			int c = (int) s.charAt(i);
			lcount[c]++;
		}
		for(int i = 0;i < t.length();i++){
			int c = (int) t.charAt(i);
			lcount[c]--;
			if(lcount[c] < 0){
				return false;
			}
		}
		
		return true;
	}

	public static void main(String args[]){
		Scanner s = new Scanner(System.in);
		String str1 = s.nextLine();
		String str2 = s.nextLine();
		if(isPermutation(str1, str2)){
			System.out.println("YES");
		}else{
			System.out.println("NO");
		}
	}
}
